core/internal/filetransfer/file_transfer_default.go

```diff
@@ -147,6 +156,8 @@ func (ft *DefaultFileTransfer) Download(task *Task) error {
 }
 
 type ProgressReader struct {
+	// Note: this turns ProgressReader into a ReadSeeker, not just a Reader!
+	// The retryablehttp client will seek to 0 on every retry.
```

Comment:
oh thats not great :( but i think we do the same in python?
anyways, maybe we can do better...?

Reply:
it would be nice to start from where we failed, if possible...

Reply:
the alternative would be to read the file into memory completely, which might not work for larger files. So I think this behavior is fine, although it took me a while to figure out because of Go's implicit interfaces! I kept rereading retryablehttp's code thinking that we were passing it a Reader and wondering why NewRequest() didn't immediately error out.

Reply:
but can we somehow keep track of the seek point and start from there instead of always seeking from 0?

Reply:
that's an interesting idea. We would need support from the backend---I think the FTP protocol supports resumed uploads, so that may be a possibility. Right now we just do an HTTP PUT, which doesn't support partial uploads I believe.